# Analysis goal 
- The goal for this project is to analyze Google Play's data to help understand what kinds of apps are likely to attract more users.
- I'll focus on free apps for this analysis.

## About the Data
As of September 2019, there were [approximately 2.8 million Android apps](https://www.statista.com/statistics/266210/number-of-available-applications-in-the-google-play-store/) on Google Play.

Collecting data for these many apps is not an easy task. So I decided to look for a data set that could help me. After some search I found two promising data sets:
- [A data set](https://www.kaggle.com/lava18/google-play-store-apps) with 10k apps collected on february 2019
- And [a data set](https://www.kaggle.com/lava18/google-play-store-apps) with 267k apps collected on april 2019


After some thought I decided to use the last one, because it has more data and was collected more recently.


## Opening the Data

In [2]:
from csv import reader

### The Google Play data set ###
opened_file = open('..\\data\\raw\\Google-Playstore-Full.csv')
read_file = reader(opened_file)
google_play = list(read_file)
google_play_header = google_play[0]
google_play_data = google_play[1:]

## Exploring the Data
To make it easier to explore the data set I created 3 functions that I will reuse throughout the project.

In [3]:
def print_header(header):
    print(header)
    print('\n')

In [4]:
def print_data(data, start = 0, end = 5):
    data_slice = data[start:end]
    for row in data_slice:
        print(row)
        print('\n')

In [5]:
def print_data_info(data):
    print('Number of rows:', len(data))
    print('Number of columns:', len(data[0]))

In [6]:
def print_data_overview():
    print_header(google_play_header)
    print_data(google_play_data)
    print_data_info(google_play_data)   

In [7]:
print_data_overview()

['App Name', 'Category', 'Rating', 'Reviews', 'Installs', 'Size', 'Price', 'Content Rating', 'Last Updated', 'Minimum Version', 'Latest Version', '', '', '', '']


['DoorDash - Food Delivery', 'FOOD_AND_DRINK', '4.548561573', '305034', '5,000,000+', 'Varies with device', '0', 'Everyone', 'March 29, 2019', 'Varies with device', 'Varies with device', '', '', '', '']


['TripAdvisor Hotels Flights Restaurants Attractions', 'TRAVEL_AND_LOCAL', '4.400671482', '1207922', '100,000,000+', 'Varies with device', '0', 'Everyone', 'March 29, 2019', 'Varies with device', 'Varies with device', '', '', '', '']


['Peapod', 'SHOPPING', '3.656329393', '1967', '100,000+', '1.4M', '0', 'Everyone', 'September 20, 2018', '5.0 and up', '2.2.0', '', '', '', '']


['foodpanda - Local Food Delivery', 'FOOD_AND_DRINK', '4.107232571', '389154', '10,000,000+', '16M', '0', 'Everyone', 'March 22, 2019', '4.2 and up', '4.18.2', '', '', '', '']


['My CookBook Pro (Ad Free)', 'FOOD_AND_DRINK', '4.647752285', '2291', 

## Data Wrangling

After a quick glance we can get some useful information about this data, like the columns that can be important ('Category', 'Rating', 'Reviews', 'Installs', 'Price', 'Content Rating').

If we pay a little more attention, we can see that the header and rows are missing the last 4 values. Since the header is mmissing too any information in these fields are meaningless. So we can start fixing that.

### Drop 4 last missing values

In [8]:
number_of_items_to_delete = 4

del google_play_header[-number_of_items_to_delete:]

for row in google_play_data:
    del row[-number_of_items_to_delete:]

In [9]:
print_data_overview()

['App Name', 'Category', 'Rating', 'Reviews', 'Installs', 'Size', 'Price', 'Content Rating', 'Last Updated', 'Minimum Version', 'Latest Version']


['DoorDash - Food Delivery', 'FOOD_AND_DRINK', '4.548561573', '305034', '5,000,000+', 'Varies with device', '0', 'Everyone', 'March 29, 2019', 'Varies with device', 'Varies with device']


['TripAdvisor Hotels Flights Restaurants Attractions', 'TRAVEL_AND_LOCAL', '4.400671482', '1207922', '100,000,000+', 'Varies with device', '0', 'Everyone', 'March 29, 2019', 'Varies with device', 'Varies with device']


['Peapod', 'SHOPPING', '3.656329393', '1967', '100,000+', '1.4M', '0', 'Everyone', 'September 20, 2018', '5.0 and up', '2.2.0']


['foodpanda - Local Food Delivery', 'FOOD_AND_DRINK', '4.107232571', '389154', '10,000,000+', '16M', '0', 'Everyone', 'March 22, 2019', '4.2 and up', '4.18.2']


['My CookBook Pro (Ad Free)', 'FOOD_AND_DRINK', '4.647752285', '2291', '10,000+', 'Varies with device', '$5.99', 'Everyone', 'April 1, 2019', 'Varies w

### Find duplicates

So here is where I hit a wall using pure python, to check for duplicates I'd have to do something like:
```python
duplicate_apps = []
unique_apps = []

for app in google_play_data:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
```
This would work just fine for a small data set. But in this data set it just takes too long, so for now I'll start using pandas

In [10]:
import pandas as pd
pd.set_option('display.float_format', lambda x: '%.5f' % x)

In [11]:
df = pd.read_csv('..\\data\\raw\\Google-Playstore-Full.csv', low_memory=False)

In [12]:
df.head()

App Name          Category  \
0                           DoorDash - Food Delivery    FOOD_AND_DRINK   
1  TripAdvisor Hotels Flights Restaurants Attract...  TRAVEL_AND_LOCAL   
2                                             Peapod          SHOPPING   
3                    foodpanda - Local Food Delivery    FOOD_AND_DRINK   
4                          My CookBook Pro (Ad Free)    FOOD_AND_DRINK   

        Rating  Reviews      Installs                Size  Price  \
0  4.548561573   305034    5,000,000+  Varies with device      0   
1  4.400671482  1207922  100,000,000+  Varies with device      0   
2  3.656329393     1967      100,000+                1.4M      0   
3  4.107232571   389154   10,000,000+                 16M      0   
4  4.647752285     2291       10,000+  Varies with device  $5.99   

  Content Rating        Last Updated     Minimum Version      Latest Version  \
0       Everyone      March 29, 2019  Varies with device  Varies with device   
1       Everyone      March 29, 2019  Varies with device  Varies with device   
2       Everyone  September 20, 2018          5.0 and up               2.2.0   
3       Everyone      March 22, 2019          4.2 and up              4.18.2   
4       Everyone       April 1, 2019  Varies with device  Varies with device   

  Unnamed: 11 Unnamed: 12 Unnamed: 13  Unnamed: 14  
0         NaN         NaN         NaN          nan  
1         NaN         NaN         NaN          nan  
2         NaN         NaN         NaN          nan  
3         NaN         NaN         NaN          nan  
4         NaN         NaN         NaN          nan

In [13]:
df.drop(columns=['Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14'], inplace=True)

### Filtering only free apps

For this we'll focus on the price column, looking closer we can see a price like this '$5.99', which indicates that the entire column is composed of strings. Knowing this we can create a function to transform the price column in number.


In [14]:
def price_to_number(price):
    price = price.replace('$', '')
    return float(price)

Let's apply this function to the price column

In [15]:
df['Price'].apply(price_to_number)

ValueError: could not convert string to float: '2.4M'

Let's check what's going on

In [16]:
df[df['Price'].str.contains(pat = 'M')]

App Name                          Category  \
13504                Never have I ever 18+                                  )   
32229               Old-time Radio presents                               NaN   
48438    Mojo Times: Bihar Hindi Video News                     Breaking News   
113151                               Steins                        Gate ALARM   
125479  2-6 Ya? E?itici �ocuk Zeka Oyunlar?                      Alfabe �?ren   
125480  2-6 Ya? E?itici �ocuk Zeka Oyunlar?                     T�rk Alfabesi   
165230              Shytter -Twitter client         not notified you follow -   
168914  CorreosTrack 2.0 (Correos de M�xico                          Mexpost)   
180371            eShagird - Online academy                      ETEA & MDCAT   
190759                    Friend in Iceland                        Tour Guide   
193869          Equalizer & Volume Booster                   Speaker Pro 2019   
194165                    Muslim Baby Names   Islamic Name Boy & Girl+Meaning   
257773                 High Volume Booster         super loud speaker booster   

                    Rating      Reviews Installs      Size Price  \
13504        GAME_STRATEGY            4        6      100+  2.4M   
32229        ENTERTAINMENT            4       20   10,000+  3.1M   
48438   NEWS_AND_MAGAZINES  4.775640965      156   10,000+  6.9M   
113151       ENTERTAINMENT  4.716867447      166      500+   67M   
125479           EDUCATION            5        1       10+   57M   
125480           EDUCATION  4.333333492       54   50,000+   43M   
165230              SOCIAL  4.098591328       71    5,000+  7.7M   
168914        PRODUCTIVITY  4.389830589       59   10,000+   16M   
180371           EDUCATION  4.504273415      117   10,000+  6.9M   
190759    TRAVEL_AND_LOCAL            5        6    1,000+   27M   
193869     MUSIC_AND_AUDIO  4.632093906      511   10,000+  2.5M   
194165           LIFESTYLE  4.388349533      927  100,000+  3.7M   
257773     MUSIC_AND_AUDIO  4.400000095       10    1,000+  3.5M   

       Content Rating Last Updated     Minimum Version Latest Version  
13504           $0.99   Mature 17+   December 30, 2018   4.0.3 and up  
32229               0     Everyone    October 16, 2018     4.1 and up  
48438               0         Teen      March 30, 2019     4.1 and up  
113151          $0.99         Teen   November 12, 2018     4.4 and up  
125479          $2.49     Everyone    October 31, 2017     2.3 and up  
125480              0     Everyone    October 31, 2017     2.3 and up  
165230              0     Everyone      March 30, 2019     4.1 and up  
168914              0     Everyone   December 21, 2018     4.1 and up  
180371              0     Everyone       March 2, 2019     4.0 and up  
190759              0     Everyone    October 16, 2017     4.0 and up  
193869              0     Everyone  September 25, 2018     2.3 and up  
194165              0     Everyone        May 23, 2018     4.0 and up  
257773              0     Everyone    November 7, 2018     4.0 and up

Some kind of shift happened to the data on these rows. As consequence of this error, we'll delete these rows.

In [17]:
#Get column indexes
indexes = df[df['Price'].str.contains(pat = 'M')].index

# Delete these row indexes from dataFrame
df.drop(indexes , inplace=True)

Let's try to apply our price_to_number function again

In [18]:
df['Price'] = df['Price'].apply(price_to_number)

ValueError: could not convert string to float: 'Varies with device'

After trying to apply our function again, we just realized that we can do a better job by just getting all rows where price is '0', this will also fix problematic rows at the same time.

In [19]:
df_free = df[df['Price'] == '0']

### Removing Duplicate Entries
Now that we have only free apps we can go back to remove duplicate entries.
First we have to think about why the duplication is happening, let's take a look at some duplicate entries.

In [20]:
df_free.sort_values('App Name', ascending=True)[df_free.duplicated(subset=['App Name'], keep=False)].head(30)

App Name             Category       Rating Reviews  \
66734                       #NAME?               SPORTS   4.55343914    4725   
85493                       #NAME?        COMMUNICATION  2.817708254     192   
65076                       #NAME?            LIFESTYLE  3.971428633    2520   
250895                      #NAME?        COMMUNICATION  4.879069805     215   
83867                       #NAME?   HEALTH_AND_FITNESS  2.846153736      39   
183900                      #NAME?              FINANCE  3.158273458     278   
209262                      #NAME?            EDUCATION  4.594594479     370   
155728                      #NAME?                TOOLS  4.647058964      85   
170318                      #NAME?            EDUCATION            5       2   
181267                      #NAME?              FINANCE  3.972222328     180   
155727                      #NAME?                TOOLS  4.673239231     355   
195447    .Net Interview Questions            EDUCATION  4.537036896      54   
28513     .Net Interview Questions            EDUCATION  4.599999905      35   
67744               10,000 Chistes        ENTERTAINMENT  4.434963703     961   
111316              10,000 Chistes            LIFESTYLE  4.361298561    6222   
9293          1000 English Stories            EDUCATION   4.59296608    7137   
108073        1000 English Stories            EDUCATION            5       2   
230351  1000 Powerful Affirmations  BOOKS_AND_REFERENCE         4.25       4   
237847  1000 Powerful Affirmations            LIFESTYLE  4.252830029     265   
227862          1000 Praises Tamil  BOOKS_AND_REFERENCE  4.719745159     157   
175808          1000 Praises Tamil            LIFESTYLE  4.810126781      79   
14242          1000+ Hindi Stories        ENTERTAINMENT  4.514175415   12698   
165690         1000+ Hindi Stories            EDUCATION  4.322033882      59   
179588           1008 Amman Pottri            LIFESTYLE  4.296360493     577   
266254           1008 Amman Pottri      MUSIC_AND_AUDIO  4.560723305     387   
168265           1008 Amman Pottri        ENTERTAINMENT  4.312063694    1003   
25768       101 Okey - ?nternetsiz           GAME_BOARD  4.543731213   68315   
17366       101 Okey - ?nternetsiz           GAME_BOARD  4.400608063   10529   
12253    10th Maths NCERT Solution  BOOKS_AND_REFERENCE   4.53656292    4335   
96589    10th Maths NCERT Solution  BOOKS_AND_REFERENCE  4.549529552    8611   

          Installs  Size Price Content Rating        Last Updated  \
66734     100,000+  4.9M     0       Everyone    November 6, 2018   
85493      10,000+   46M     0           Teen    October 31, 2018   
65076     100,000+   16M     0       Everyone   February 25, 2019   
250895      5,000+   16M     0       Everyone   December 10, 2018   
83867       1,000+  4.3M     0       Everyone        May 14, 2015   
183900     50,000+  3.0M     0       Everyone       March 4, 2019   
209262     10,000+   18M     0       Everyone   December 12, 2018   
155728      1,000+  5.2M     0       Everyone      March 26, 2019   
170318        100+  6.4M     0       Everyone       July 23, 2018   
181267     10,000+  3.1M     0       Everyone       March 4, 2019   
155727      5,000+  3.3M     0       Everyone      March 25, 2019   
195447     10,000+   20M     0       Everyone    January 14, 2018   
28513       5,000+  5.3M     0       Everyone   November 26, 2018   
67744     100,000+  4.2M     0       Everyone    December 2, 2018   
111316    500,000+  4.4M     0       Everyone      August 8, 2018   
9293    1,000,000+  4.4M     0       Everyone       March 3, 2019   
108073      1,000+  4.0M     0       Everyone   December 29, 2018   
230351        100+  4.9M     0           Teen   December 20, 2018   
237847    100,000+  3.2M     0   Everyone 10+    November 8, 2018   
227862     10,000+  1.9M     0       Everyone       April 4, 2015   
175808      5,000+   63M     0       Everyone      April 11, 2017   
14242     500,000+   12M   

As we can the first apps are missing names and they were just replaced with '#NAME?'. Let's remove them.

In [21]:
#Get column indexes
indexes = df_free[df_free['App Name'] == '#NAME?'].index

# Delete these row indexes from dataFrame
df_free.drop(indexes , inplace=True)

Now to avoid removing duplicates randomly, we can remove them based on some information like rating, number of reviews, number of installs, last updated or last version. But number of reviews seems the best, the highest number of reviews among the duplicates would give us at the same time the most reliable rating, so we'll keep the highest number of reviews.

In [22]:
#First we'll sort the data frame based on the Reviews column
df_free.sort_values('Reviews', ascending=False, inplace=True)

In [23]:
#After we'll drop the duplicated names leaving only the highest number of reviews app
df_free.drop_duplicates(subset='App Name', keep='first', inplace=True)

## Most Common Apps by Category

In [24]:
df_free_groupby_category_percentage = df_free.groupby('Category', as_index=False).agg({'App Name': lambda x: (x.count()/len(df_free)) * 100}).rename(columns={'App Name':'Percentage'})
df_free_groupby_category_percentage.sort_values('Percentage', ascending=False)

Category  Percentage
8             EDUCATION    12.46185
9         ENTERTAINMENT     7.92797
45                TOOLS     7.81738
3   BOOKS_AND_REFERENCE     7.56104
36      MUSIC_AND_AUDIO     6.84604
33            LIFESTYLE     5.56865
4              BUSINESS     4.12152
11              FINANCE     4.05807
39      PERSONALIZATION     3.79788
41         PRODUCTIVITY     3.44210
37   NEWS_AND_MAGAZINES     3.14590
30   HEALTH_AND_FITNESS     2.85870
47     TRAVEL_AND_LOCAL     2.64994
40          PHOTOGRAPHY     2.58607
44               SPORTS     2.17371
6         COMMUNICATION     2.16171
42             SHOPPING     2.08369
43               SOCIAL     1.81021
34  MAPS_AND_NAVIGATION     1.58774
35              MEDICAL     1.34383
22          GAME_PUZZLE     1.26968
12       FOOD_AND_DRINK     1.23152
48        VIDEO_PLAYERS     1.00648
19          GAME_CASUAL     0.91132
15          GAME_ARCADE     0.87274
1     AUTO_AND_VEHICLES     0.82430
20     GAME_EDUCATIONAL     0.69528
0        ART_AND_DESIGN     0.68370
49              WEATHER     0.60655
25      GAME_SIMULATION     0.54739
13          GAME_ACTION     0.54311
17            GAME_CARD     0.38793
2                BEAUTY     0.36993
31       HOUSE_AND_HOME     0.36007
29            GAME_WORD     0.34378
16           GAME_BOARD     0.31249
24    GAME_ROLE_PLAYING     0.30220
14       GAME_ADVENTURE     0.30092
23          GAME_RACING     0.29577
27        GAME_STRATEGY     0.28677
26          GAME_SPORTS     0.28377
28          GAME_TRIVIA     0.27520
10               EVENTS     0.25248
32   LIBRARIES_AND_DEMO     0.23319
38            PARENTING     0.23062
5                COMICS     0.19718
18          GAME_CASINO     0.14917
7                DATING     0.12945
21           GAME_MUSIC     0.09216
46               TRAVEL     0.00043

As we can see there are more than one category for games so lets put them together.

In [25]:
df_free_groupby_category_percentage[df_free_groupby_category_percentage['Category'].str.contains("GAME")].sum()

Category      GAME_ACTIONGAME_ADVENTUREGAME_ARCADEGAME_BOARD...
Percentage                                              7.86967
dtype: object

Google Play shows a balanced landscape of apps. Even if we group up all games categories they would come in third place. Now we'd like to get an idea about the kind of apps that have most users.

## Most Popular Apps by Category

One way to find out what categories are the most popular (have the most users) is to calculate the average number of installs for each app category. We can find this information in the Installs column. However, the install numbers don't seem precise enough, we have numbers like '50+', '100+', '100,000+'. But for our purposes we don't need to be so precise, we need only to have an idea of which category atracts more users. For that we'll only convert these numbers to float, so we can make calculations.

In [26]:
df_free['Installs'] = df_free['Installs'].apply(lambda x: float(x.replace(',', '').replace('+', '')))

With the code above we remove commas and plus signs and convert the string to float so we can make some calculations.

Let's get the average of number of installs per category.

In [27]:
df_free.groupby('Category', as_index=False)['Installs'].mean().sort_values('Installs', ascending=False)

Category      Installs
23          GAME_RACING 6222893.08406
13          GAME_ACTION 4724329.77269
26          GAME_SPORTS 4166148.82175
48        VIDEO_PLAYERS 3773136.82155
27        GAME_STRATEGY 3425426.62332
15          GAME_ARCADE 3296286.01473
19          GAME_CASUAL 2937325.03293
21           GAME_MUSIC 2644452.86047
6         COMMUNICATION 2596150.06365
25      GAME_SIMULATION 2254367.49413
18          GAME_CASINO 1828752.19828
14       GAME_ADVENTURE 1708987.13248
43               SOCIAL 1644402.58915
24    GAME_ROLE_PLAYING 1608265.53191
41         PRODUCTIVITY 1563329.75455
40          PHOTOGRAPHY 1398672.42732
22          GAME_PUZZLE 1346160.14078
47     TRAVEL_AND_LOCAL 1308115.61825
45                TOOLS 1218567.92318
16           GAME_BOARD 1080260.50754
29            GAME_WORD  963137.23441
20     GAME_EDUCATIONAL  726924.91060
42             SHOPPING  661821.71467
17            GAME_CARD  652852.79669
28          GAME_TRIVIA  585626.08723
49              WEATHER  533323.83392
37   NEWS_AND_MAGAZINES  499898.59482
7                DATING  488803.94702
39      PERSONALIZATION  423701.32698
5                COMICS  417495.11957
9         ENTERTAINMENT  383384.34307
31       HOUSE_AND_HOME  360268.76310
36      MUSIC_AND_AUDIO  328076.25045
34  MAPS_AND_NAVIGATION  314267.62959
30   HEALTH_AND_FITNESS  297285.66937
0        ART_AND_DESIGN  227266.66708
11              FINANCE  223438.34351
12       FOOD_AND_DRINK  198810.44831
1     AUTO_AND_VEHICLES  196561.96152
32   LIBRARIES_AND_DEMO  179525.42463
44               SPORTS  161077.99684
38            PARENTING  153948.06691
33            LIFESTYLE  152758.93111
2                BEAUTY  149071.46118
4              BUSINESS  139258.62350
3   BOOKS_AND_REFERENCE  105154.89688
8             EDUCATION   77019.27349
35              MEDICAL   63612.10080
10               EVENTS   47432.08319
46               TRAVEL   10000.00000

As we can see it's obvious that games are popular.

In [35]:
df_free[df_free['Category'] == 'VIDEO_PLAYERS'].sort_values('Installs', ascending=False).head(20)

App Name       Category  \
813                                               YouTube  VIDEO_PLAYERS   
6412                              Google Play Movies & TV  VIDEO_PLAYERS   
3154            LIKE Video -Magic Video Maker & Community  VIDEO_PLAYERS   
10796                                   Motorola FM Radio  VIDEO_PLAYERS   
8987    VideoShow Video Editor, Video Maker, Beauty Ca...  VIDEO_PLAYERS   
8986    DU Recorder � Screen Recorder, Video Editor, Live  VIDEO_PLAYERS   
2114                                      VLC for Android  VIDEO_PLAYERS   
3151    VMate 2019- Video Downloader &Best Video Tube ...  VIDEO_PLAYERS   
3497                                     Motorola Gallery  VIDEO_PLAYERS   
44563              VivaVideo - Video Editor & Photo Movie  VIDEO_PLAYERS   
8718                                Samsung Video Library  VIDEO_PLAYERS   
112157                      �Torrent�- Torrent Downloader  VIDEO_PLAYERS   
14401                          Peel Smart Remote TV Guide  VIDEO_PLAYERS   
190161             Dubsmash - Dance Videos & Lip Sync App  VIDEO_PLAYERS   
8978    Quik � Free Video Editor for photos, clips, music  VIDEO_PLAYERS   
13025         MiniMovie - Free Video and Slideshow Editor  VIDEO_PLAYERS   
1273                                                 Vine  VIDEO_PLAYERS   
109434               Kwai - Short Video Maker & Community  VIDEO_PLAYERS   
253120     Magisto - Video Editor & Music Slideshow Maker  VIDEO_PLAYERS   
8599    PowerDirector - Video Editor App, Best Video M...  VIDEO_PLAYERS   

             Rating   Reviews         Installs                Size Price  \
813      4.36842823  41919102 5000000000.00000  Varies with device     0   
6412    3.703356266   1048972 1000000000.00000  Varies with device     0   
3154    4.581243515   2048739  100000000.00000                 45M     0   
10796   3.924235821     58141  100000000.00000  Varies with device     0   
8987    4.559087276   4458794  100000000.00000                 22M     0   
8986    4.745390892   3992518  100000000.00000                 17M     0   
2114    4.423864365   1139749  100000000.00000  Varies with device     0   
3151    4.248163223    323372  100000000.00000                 17M     0   
3497    3.947426319    126318  100000000.00000                 23M     0   
44563   4.603675365  10855572  100000000.00000                 40M     0   
8718    4.466049671     39661  100000000.00000                5.8M     0   
112157  4.542450905   3800761  100000000.00000  Varies with device     0   
14401   4.295152664    658432  100000000.00000  Varies with device     0   
190161  4.228127003   1997744  100000000.00000                 11M     0   
8978    4.738492489    975119   50000000.00000                 99M     0   
13025   4.518614292    509444   50000000.00000  Varies with device     0   
1273    4.186682224   1139574   50000000.00000                1.3M     0   
109434  4.449896336    419552   50000000.00000                 32M     0   
253120  4.285580158   1008323   50000000.00000                 51M     0   
8599    4.512517929    898860   50000000.00000                 54M     0   

       Content Rating       Last Updated     Minimum Version  \
813              Teen      April 2, 2019  Varies with device   
6412             Teen      April 1, 2019  Varies with device   
3154             Teen      April 2, 2019          4.0 and up   
10796        Everyone  December 13, 2018  Varies with device   
8987         Everyone      April 3, 2019        4.0.3 and up   
8986         Everyone      April 3, 2019          5.0 and up   
2114         Everyone      April 1, 2019          2.3 and up   
3151             Teen      April 1, 2019        4.0.3 and up   
3497         Everyone   January 25, 2016  Varies with device   
44563            Teen      April 3, 2019          4.1 and up   
8718         Everyone     March 28, 2019          7.0 and up   
112157       Everyone      April 1, 2019          4.1 and up   
14401        Everyone  

In [34]:
df_free[df_free['Category'] == 'COMMUNICATION'].sort_values('Installs', ascending=False).head(20)

App Name       Category  \
632              Messenger � Text and Video Chat for Free  COMMUNICATION   
815                         Skype - free IM & video calls  COMMUNICATION   
3267                             Samsung Internet Browser  COMMUNICATION   
842                          Google Chrome: Fast & Secure  COMMUNICATION   
1981                                             Hangouts  COMMUNICATION   
2724                                                Gmail  COMMUNICATION   
6411                Google Duo - High Quality Video Calls  COMMUNICATION   
671                                    WhatsApp Messenger  COMMUNICATION   
832       Viber Messenger - Messages, Group Chats & Calls  COMMUNICATION   
2729    UC Browser � Short Video Status & Video Downlo...  COMMUNICATION   
1954                          LINE: Free Calls & Messages  COMMUNICATION   
154068                                               Azar  COMMUNICATION   
3265                                             Contacts  COMMUNICATION   
3016                          BBM - Free Calls & Messages  COMMUNICATION   
2737                                             Telegram  COMMUNICATION   
685                 Messenger Lite: Free Calls & Messages  COMMUNICATION   
2747                                                  Kik  COMMUNICATION   
2741                         Yahoo Mail � Organized Email  COMMUNICATION   
2736    UC Browser Mini - Video Downloader & Video Status  COMMUNICATION   
3800                                     Carrier Services  COMMUNICATION   

             Rating   Reviews         Installs                Size Price  \
632     4.085856438  65469531 1000000000.00000  Varies with device     0   
815     4.134609699  10746013 1000000000.00000  Varies with device     0   
3267    4.424014568    832714 1000000000.00000  Varies with device     0   
842     4.335205078  13636591 1000000000.00000  Varies with device     0   
1981    4.040542603   3960560 1000000000.00000  Varies with device     0   
2724    4.346980095   5614163 1000000000.00000  Varies with device     0   
6411    4.596403599   3641252 1000000000.00000                 20M     0   
671     4.417610168  86214292 1000000000.00000  Varies with device     0   
832     4.362102032  12723787  500000000.00000  Varies with device     0   
2729    4.450339317  19573637  500000000.00000                 44M     0   
1954    4.185222626  11390281  500000000.00000                 94M     0   
154068  4.213160515   1256622  100000000.00000  Varies with device     0   
3265    4.348867416    152998  100000000.00000  Varies with device     0   
3016    4.274343967  12958310  100000000.00000                 45M     0   
2737    4.393939018   3532906  100000000.00000  Varies with device     0   
685     4.410111427   2421586  100000000.00000  Varies with device     0   
2747    4.280104637   2467050  100000000.00000  Varies with device     0   
2741    4.346459389   4549487  100000000.00000  Varies with device     0   
2736    4.408800125   4005437  100000000.00000                8.8M     0   
3800    4.402204514    144168  100000000.00000  Varies with device     0   

       Content Rating       Last Updated     Minimum Version  \
632          Everyone      April 2, 2019  Varies with device   
815          Everyone     March 28, 2019  Varies with device   
3267         Everyone  February 21, 2019          5.0 and up   
842          Everyone     March 26, 2019  Varies with device   
1981         Everyone     March 29, 2019  Varies with device   
2724         Everyone      April 2, 2019  Varies with device   
6411         Everyone      April 2, 2019          4.4 and up   
671          Everyone     March 27, 2019  Varies with device   
832          Everyone      April 2, 2019          4.1 and up   
2729             Teen      April 2, 2019          4.0 and up   
1954         Everyone     March 20, 2019          4.4 and up   
154068     Mature 17+     March 13, 2019  Varies with device   
3265         Everyone  

In [33]:
df_free[df_free['Category'] == 'SOCIAL'].sort_values('Installs', ascending=False).head(20)

App Name Category  \
831                                         Facebook Lite   SOCIAL   
653                                             Instagram   SOCIAL   
704                                              Facebook   SOCIAL   
642                                                TikTok   SOCIAL   
691                                              Snapchat   SOCIAL   
13915    Hike News & Content (for chatting go to new app)   SOCIAL   
7637                        VK � social network and calls   SOCIAL   
1991                        Tango - Live Video Broadcasts   SOCIAL   
463                                             Pinterest   SOCIAL   
2710    LinkedIn: Jobs, professional profile, & networ...   SOCIAL   
1962                                               Tumblr   SOCIAL   
20592   ShareChat - Make Friends, WhatsApp Status & Vi...   SOCIAL   
1995                       Badoo - Free Chat & Dating App   SOCIAL   
102032                     Vigo Video - Funny Short Video   SOCIAL   
3158      BIGO LIVE - Live Stream, Live Video & Live Chat   SOCIAL   
32732                                                  OK   SOCIAL   
1978                          SKOUT - Meet, Chat, Go Live   SOCIAL   
1960                                               LOVOO�   SOCIAL   
35324    LiveMe - Video chat, new friends, and make money   SOCIAL   
1985                                  POF Free Dating App   SOCIAL   

             Rating   Reviews         Installs                Size Price  \
831     4.288809299  10866006 1000000000.00000  Varies with device     0   
653      4.51955986  79726403 1000000000.00000  Varies with device     0   
704     4.087946415  85766433 1000000000.00000  Varies with device     0   
642     4.386767387   9670607  500000000.00000                 72M     0   
691      4.07791853  19026060  500000000.00000                 56M     0   
13915   4.337181568   2929521  100000000.00000  Varies with device     0   
7637    3.821457386   6286733  100000000.00000                 51M     0   
1991    4.290686131   3822103  100000000.00000                 80M     0   
463     4.617506027   5064395  100000000.00000  Varies with device     0   
2710    4.240080833   1372067  100000000.00000                 23M     0   
1962    4.345463276   3241472  100000000.00000  Varies with device     0   
20592   4.521960258    950124  100000000.00000                 42M     0   
1995    4.299030304   4216266  100000000.00000  Varies with device     0   
102032  4.326038837    613004  100000000.00000                 68M     0   
3158    4.460019588   2968932  100000000.00000                 34M     0   
32732    4.36797905   2267408  100000000.00000  Varies with device     0   
1978    4.257332802   1109664   50000000.00000                 72M     0   
1960    4.026992321    906584   50000000.00000  Varies with device     0   
35324   4.444291115    524118   50000000.00000  Varies with device     0   
1985    4.135750771   1318838   50000000.00000  Varies with device     0   

       Content Rating       Last Updated     Minimum Version  \
831              Teen     March 28, 2019  Varies with device   
653              Teen      April 2, 2019  Varies with device   
704              Teen      April 2, 2019  Varies with device   
642              Teen      April 2, 2019          4.1 and up   
691              Teen      April 3, 2019          4.4 and up   
13915            Teen  February 15, 2019  Varies with device   
7637       Mature 17+      April 2, 2019  Varies with device   
1991       Mature 17+      April 2, 2019          4.4 and up   
463              Teen      April 2, 2019  Varies with device   
2710         Everyone     March 28, 2019          5.0 and up   
1962       Mature 17+     March 27, 2019  Varies with device   
20592            Teen      April 1, 2019          4.1 and up   
1995       Mature 17+      April 2, 2019  Varies with device   
102032           Teen      April 1, 2019        4.0.3 and up   
3158             Teen

To support even more games popularity, categories like video players, communication and social are dominated by apps like YouTube, Skype and Facebook with billions of installs each, making these categories seems more popular than they really are.

## Conclusion
In this project, we analyzed data about the Google Play mobile apps with the goal of recommending an app profile that can be profitable.

After the analysis, we concluded that making a game would probably be the best bet (racing games are more popular, but we think that any kind of game would be fine), they are popular and people tend to install multiple games of the same genre, unlike other categories of apps that people generally have a preference and that we would start at a disadvantage in competition.